app/src/main/java/com/rootstrap/android/ui/activity/main/targetpoint/MapFragment.kt

```diff
@@ -44,10 +52,39 @@ class MapFragment : Fragment(), OnMapReadyCallback {
         mapFragment.getMapAsync(this)
     }
 
-    // TODO: remove example marker when we have the user location
     override fun onMapReady(googleMap: GoogleMap) {
         mMap = googleMap
-        addMarker(EXAMPLE_LAT, EXAMPLE_LONG)
+        checkLocationPermission()
+    }
+
+    private fun checkLocationPermission() {
+        if (requireContext().checkNotGrantedPermissions(locationPermissions).isEmpty()) {
+            getDeviceLocation()
+        } else {
+            listener?.askForLocationPermission {
+                getDeviceLocation()
+            }
+        }
+    }
+
+    private fun getDeviceLocation() {
+        try {
+            mMap.isMyLocationEnabled = true
+            fusedLocationClient.lastLocation.addOnSuccessListener { location ->
+                location?.run {
+                    addMarker(latitude, longitude)
+                }
+            }
+        } catch (e: SecurityException) {
+            e.printStackTrace()
+        }
+    }
```

Comment:
I'd create a LocationManager, to manage everything related to the user location(updates etc..), in case you needed it in another place.

Reply:
Thank you, I'd add it in the next PR